File: Design-Patterns-Using-Swift/Proxy/ProxyTest.swift
import Foundation

// Subject protocol
protocol Subject {
    func request()
}

// RealSubject class conforming to Subject protocol
class RealSubject: Subject {
    func request() {
        print("Concrete Subject Request Method")
    }
}

// Proxy class conforming to Subject protocol
class Proxy: Subject {
    private var realSubject: RealSubject

    init() {
        self.realSubject = RealSubject()
    }

    func request() {
        realSubject.request()
    }
}

// Client Code
let proxy = Proxy()
proxy.request()

/*
Concrete Subject Request Method
*/

File: Design-Patterns-Using-Kotlin/Proxy/ProxyTest.kt
interface Subject {
    fun request()
}

class RealSubject : Subject {
    override fun request() {
        println("Concrete Subject Request Method")
    }
}

class Proxy : Subject {
    private val realSubject = RealSubject()

    override fun request() {
        realSubject.request()
    }
}

// Client Code
fun main() {
    val proxy = Proxy()
    proxy.request()
}

/*
Concrete Subject Request Method
*/